### PR TITLE
feat: added weight decay

### PR DESCRIPTION
**Pull request by lucasosiewicz**

### CODE REVIEW

**Comment by rojberr** [comment]
Started a GPU training pipeline: https://github.com/GHOST-Science-Club/tree-classification-irim/actions/runs/14407640186
and will try to build it on PUT server. Results can be seen in WanDb.

**Comment by rojberr** [comment]
> dataset:   folder: "src/data"   species_folders:     Abies_alba: "data/imagery-Abies_alba.zip"     Abies_nordmanniana: "data/imagery-Abies_nordmanniana.zip"     Castanea_sativa: "data/imagery-Castanea_sativa.zip"     Fagus_sylvatica: "data/imagery-Fagus_sylvatica.zip"     Larix_decidua: "data/imagery-Larix_decidua.zip"     Picea_abies: "data/imagery-Picea_abies.zip"     Pinus_halepensis: "data/imagery-Pinus_halepensis.zip"     Pinus_nigra: "data/imagery-Pinus_nigra.zip"     Pinus_nigra_laricio: "data/imagery-Pinus_nigra_laricio.zip"     Pinus_pinaster: "data/imagery-Pinus_pinaster.zip"     Pinus_sylvestris: "data/imagery-Pinus_sylvestris.zip"     Pseudotsuga_menziesii: "data/imagery-Pseudotsuga_menziesii.zip"     Quercus_ilex: "data/imagery-Quercus_ilex.zip"     Quercus_petraea: "data/imagery-Quercus_petraea.zip"     Quercus_pubescens: "data/imagery-Quercus_pubescens.zip"     Quercus_robur: "data/imagery-Quercus_robur.zip"     Quercus_rubra: "data/imagery-Quercus_rubra.zip"     Robinia_pseudoacacia: "data/imagery-Robinia_pseudoacacia.zip"   main_subfolders:     aerial_imagery: "imagery/"     lidar: "lidar/"
> 
> model:   name: "resnet18" # currently supporting resnet18, vit and inception_v3
> 
> training:   batch_size: 32   learning_rate: 0.0001   max_epochs: 100   freeze: true   weight_decay: 0.0001      oversample:     oversample_factor: 4     oversample_threshold: 1000   # undersample:   #   target_size: 530
> 
>   dataloader:     auto: true     num_workers: 0     pin_memory: false     presistent_workers: false
> 
>   early_stopping:     apply: true     monitor: "val_loss"     patience: 3     mode: "min"
> 
> device: "gpu"
> 
> 

So proud of this small feature 🙈

**Review by addobosz — APPROVED** [review]
gj

**Review by rojberr — APPROVED** [review]
